mgmt/tribe/worker/worker.go

```diff
@@ -324,55 +324,68 @@ func (w worker) loadPlugin(plugin core.Plugin) error {
 			}).Info("unable to create client")
 			continue
 		}
-		resp, err := c.TribeRequest()
+		f, err := w.downloadPlugin(c, plugin)
+		// If we can't download from this member, try the next
 		if err != nil {
-			logger.WithFields(log.Fields{
-				"err": err,
-				"url": url,
-			}).Info("plugin not found")
+			logger.Error(err)
 			continue
 		}
-		if resp.StatusCode == 200 {
-			if resp.Header.Get("Content-Type") != "application/x-gzip" {
-				logger.WithField("content-type", resp.Header.Get("Content-Type")).Error("Expected application/x-gzip")
-			}
-			dir, err := ioutil.TempDir("", "")
-			if err != nil {
-				logger.Error(err)
-				return err
-			}
-			f, err := os.Create(path.Join(dir, fmt.Sprintf("%s-%s-%d", plugin.TypeName(), plugin.Name(), plugin.Version())))
-			if err != nil {
-				logger.Error(err)
-				f.Close()
-				return err
-			}
-			io.Copy(f, resp.Body)
-			f.Close()
-			err = os.Chmod(f.Name(), 0700)
-			if err != nil {
-				logger.Error(err)
-				return err
-			}
-			rp, err := core.NewRequestedPlugin(f.Name())
-			if err != nil {
-				logger.Error(err)
-				return err
-			}
-			_, err = w.pluginManager.Load(rp)
-			if err != nil {
-				logger.Error(err)
-				return err
-			}
-			if w.isPluginLoaded(plugin.Name(), plugin.TypeName(), plugin.Version()) {
-				return nil
-			}
-			return errors.New("failed to load plugin")
+		rp, err := core.NewRequestedPlugin(f.Name())
+		if err != nil {
+			logger.Error(err)
+			return err
+		}
+		_, err = w.pluginManager.Load(rp)
+		if err != nil {
+			logger.Error(err)
+			return err
+		}
+		if w.isPluginLoaded(plugin.Name(), plugin.TypeName(), plugin.Version()) {
```

Comment:
add a comment. Making sure the plugin is loaded. Otherwise, I thought it should check this in the beginning before loading a plugin. Then I saw there is a similar check at the beginning. If no error is not good enough when loading a plugin?